src/cirrus/lib2/process_payload.py

```diff
@@ -22,10 +25,38 @@
 
 
 # clients
+_event_manager = None
 _statedb = None
 _stepfunctions = None
 
 
+def get_event_manager():
+    global _event_manager
+    if _event_manager is None:
+        _event_manager = WorkflowEventManager(logger=logger, statedb=get_statedb())
+    return _event_manager
+
+
+class EventManagerFlush(ContextDecorator):
```

Comment:
This was kinda my idea with the lambda handler decorator...but it still feels like you are only using this within the bounds of a single call stack, so you could easily pass a wfem instance in.